CONTRIBUTING.md

```diff
@@ -38,6 +50,8 @@ If the pull request is simple and short lived, it can be initialized with review
 If the pull request is more complex and is being developed over time, it may be benefitial to start it in a `Draft` state.
 This allows other contributors to monitor the progress and volunteer feedback while annotating that the pull request is not yet ready for review.
 
+If a pull request is particularly large in scope, consider reducing the scope of the pull request and moving work to the `experimental/` directory, as discussed above.
+
```

Comment:
`suggestion (optional, non-blocking)`: "and either mark the unfinished parts behind experimental flag or place the code in experimental/ directory if possible".